.github/workflows/release-docker-image-demo.yml

```diff
@@ -0,0 +1,81 @@
+name: Release process
+
+on: push
+
+env:
+  LONG_COMMIT_HASH: 058650a33d36df49cde582cb480666c080af3d64
+  SHORT_COMMIT_HASH: 058650a
+  SEMVER_VERSION: 0.0.1
+
+jobs:
+  create-release:
+    runs-on: ubuntu-24.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          ref: ${{env.LONG_COMMIT_HASH}}
+
+      - name: Set up JDK 21
+        uses: actions/setup-java@v4
+        with:
+          java-version: 21
+          distribution: 'oracle'
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Set new snapshot version
+        run: mvn build-helper:parse-version versions:set -DnewVersion=${{ env.SEMVER_VERSION }}-SNAPSHOT -DgenerateBackupPoms=false
+
+      - name: Build JAR with mvn package
+        run: mvn clean package -Dtest=\!CypressE2ETest,\!DockerCypressE2ETest
```

Comment:
This should happen before the release and not during the release. Take a look at [the Delivery Pipeline Concept](https://github.com/puzzle/delivery-pipeline-concept).